### ISSUE TITLE
ACF Blocks: as of WordPress 5.5, the $is_preview parameter is now of type WP_Block instead of boolean (only in front-end)

### ISSUE THREAD

**Issue by first-wgrt** [AdvancedCustomFields/acf#353]
After updating to WP 5.5, I noticed that the third parameter `$is_preview` in the `render_callback` and `render_template` now is a `WP_Block` instance instead of a boolean.

The problem does not occur inside the block editor, only on the front end.

This might break many sites using ACF Blocks that update to WP 5.5:
```php
<?php

if($is_preview):
    // Editor preview HTML
else:
    // Front end block HTML
endif;
```

I have done this many times, and it works fine in WP < 5.5, but now `$is_preview` always evaluates to `true` in the front end. That means my editor preview HTML is also shown on the front end...

**Comment by justinkruit** [comment]
Same here, for some blocks I don't make previews (too complex or not practical), and now it's showing the placeholders on the entire website.

**Comment by studiorubenz** [comment]
thanks @first-wgrt for reporting this, I am running into the same issue and I can confirm your observation of the `render_callback` function returning an object instead of a boolean.

As a quick fix / dirty hack I am overriding the `$is_preview` argument with the native WP function:

```php
function my_acf_block_render_callback( $block, $content = '', $is_preview = false ) {
    // ...
    // TODO: SHOULD BE
    // $context['is_preview'] = $is_preview;
   
    // temporarily changed to
    $context['is_preview'] = is_preview(); 
    // ...
```

**Comment by elliotcondon** [comment]
Hi all,

Thanks for the bug report,
Elliot here - ACF dev.

I've got this issue on my to-do for today and will hope to release an update shortly :)

**Comment by elliotcondon** [comment]
Update: We have released a 5.8.14 update to take care of this bug - thanks again for your help!​​

**Comment by first-wgrt** [comment]
Wow. Thanks for the quick fix. I can confirm the problem is solved for me after updating to 5.8.14. Closing this.
